src/buf.c

```diff
@@ -266,4 +285,10 @@ struct Buf *buf_append (struct Buf *buf, const void *ptr, int n_elem)
 	return buf;
 }
 
+struct Buf *buf_append_alloc (struct Buf *buf, const void *ptr, int n_elem)
+{
+	buf_append(&alloc_buf, ptr, n_elem);
```

Comment:
It is more that the contents of a `Buf` of `char*` can be a mixed bag of allocated vs const strings.

While `Buf` is somewhat like a std::vector, `Buf`'s interface only allows for appending entries, not for removing or changing them.

I was careful to limit alloc_buf use to buf.c, so the scope is actually file (though I did forget to set it static).
This actually seems like a very normal pattern if you think of buf_init_alloc as initializing the buf interface
buf_destroy_alloc as finalizing.

What you suggest will involve either altering the `Buf` structure to contain its own list of free-able elements,
or require every element be free-able (come from a strdup, etc.).  In the former case, there would still need to be buf_append_alloc.  In the latter case, we would either need change `Buf` to know about the element type (currently it just knows about the element size) maybe with an enum, or there would need to be a separate variant of buf_destroy() and if you want the interface to take care of the strdup then we still need buf_append_alloc.